test/index.js

```diff
@@ -381,6 +381,68 @@ test('invalid union value passed to "type" option', function(t) {
   t.end();
 });
 
+// Test strict mode
+
+test('unknown long option --bar', function(t) {
+  const passedArgs = ['--foo', '--bar'];
```

Comment:
Should we default `strict` to `true`, like @ljharb has advocated?

@shadowspawn, @aaronccasanova what are your votes?

Reply:
Also @bakkot, i believe.

Reply:
https://github.com/pkgjs/parseargs/issues/11#issuecomment-1082664767

Reply:
Yes, I vote `strict:true` by default. I like the idea of having `parseArgs` handle a majority cases off the shelf and allowing advance users/and library authors to explicitly opt out in favor of their own validation.

Reply:
The one ergonomic thing I don't like about strict by default, is that you end up having empty configuration:

```js
parseArgs({
  strict: true,
  options: {
    foo: {},
  },
});
```

If you decide you want the default `boolean` behavior.

What if we just don't make`type` optional. I prefer this because you also don't need to read docs to remember if `type` defaults to `string` or `boolean` (which I honestly forget).

Reply:
> What if we just don't make `type` optional. I prefer this because you also don't need to read docs to remember if `type` defaults to `string` or `boolean` (which I honestly forget).

That's sensible to me. Similar to the behavior being `strict` for users the configuration would be `strict` for authors.

Reply:
I agree making type required is sensible from a readability point of view. Outside scope of this PR though?

Reply:
Feel free to reopen this conversation, but closing now as we all seem to be in agreement about `strict:true` default and explicitly setting option config types.